src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -680,11 +691,31 @@ private bool ShouldSkip(ReplicationBatchItem item, OutgoingReplicationStatsScope
 
                     break;
 
-                case AttachmentReplicationItem _:
+                case AttachmentReplicationItem attachment:
```

Comment:
maybe simpler would be just to check if the item's etag is the lowest etag for this hash? 
This would mean it is the first time we encounter this stream, so we should send it.
 
Any other case would be handled by the `MissingAttachment` if needed

Reply:
That might be a good idea, yes. I'll test it.

Reply:
No, there is no actual etag assigned to the stream, and while we have an index on `Hash`, it does not include the etag, so it isn't meaningfully sorted for our needs.

Reply:
Not on the stream but on the `AttachmentReplicationItem`.
We can get all the item with the given hash and know if we have the lowest etag, no?

Reply:
We _can_, but they aren't sorted. In this case, we'll have to scan through all of the entries each time. Not fun.
In the scenario we are talking about, there are avg of 14 attachments per stream, but it makes a lot of sense to have a stream that has thousands of attachments (think: profile picture with a default one).

I would rather avoid this sort of thing.

Reply:
Maybe we can add an index (hash+etag) which will give us what we need?
(This will require a schema upgrade)

Reply:
> Maybe we can add an index (hash+etag) which will give us what we need? (This will require a schema upgrade)

Adding extra index and doing schema upgrade to optimize the case that we see for the first time doesn't seem to be worth it IMO.

Reply:
That is my thinking as well